src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Remote/RemoteResponse.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Runtime.Serialization;
+
+namespace Microsoft.CodeAnalysis.Razor.Remote;
+
+[DataContract]
+internal record struct RemoteResponse<T>(
+    [property: DataMember(Order = 0)] bool StopHandling,
+    [property: DataMember(Order = 1)] T Result)
```

Comment:
I love elevating this concept so that any service can use it. I have a few ideas and thoughts that you can take or leave:

1. Consider whether `Data` or something else is a better name. To me, the words `Result` and `Response` are bit too close semantically and are pretty much interchangeable.
2. Is it possible to add a `T : class` constraint?
3. It might be helpful to add `HasData` and/or `TryGetData(...)` APIs. If `T` is constrained to `class`, declare `Data` as `T?` and then add `[MemberNotNullWhen(true, Data)]` to `HasData` and `[NotNullWhen(true)]` to `TryGetData`.
4. `StopHandling` feels a little awkward to "handle" on the client side. I wonder if there's a better name or combination of signals. In general, it feels strange that the remote service would have so much knowledge about what the client does with the response. It also feels a bit odd to me that a `RemoteResponse` might have a `Result` and `StopHandling == false`.  I don't have any great suggestions, so please don't make any changes related to this unless you have better ideas. However, I suggest that we keep thinking on this logic.

Reply:
I have no problem with merging this as is. I just wanted to write my thoughts down. We can definitely get back to them later as we go.